core/src/test/java/apoc/index/SchemaIndexTest.java

```diff
@@ -86,76 +102,191 @@ public void testDistinctPropertiesOnFirstIndex() {
+                    extractedPerson(result);
+                    extractedSchemaTest(result);
+                    assertFalse(result.hasNext());
+                });
+
+        testResult(db,SCHEMA_DISTINCT_COUNT_ORDERED,
+                map("label", SCHEMA_LABEL, "key",""),
+                (result) -> {
+                    extractedSchemaTest(result);
+                    assertFalse(result.hasNext());
+                });
+
+        db.executeTransactionally("drop index SchemaTest");
     }
 
     @Test(timeout = 5000L)
-    public void testDistinctWithMultiLabelFullTextIndexShouldNotHangs() throws Exception {
-        db.executeTransactionally("CREATE FULLTEXT INDEX fulltextLabel1 FOR (n:Label1|Label2) ON EACH [n.prop1,n.prop3]");
-        // insert a non-fulltext index, with the same label and prop
-        db.executeTransactionally("CREATE RANGE INDEX FOR (n:One) ON (n.prop1)");
+    public void testDistinctWithFullTextIndexShouldNotHangs() throws Exception {
+        db.executeTransactionally("CREATE FULLTEXT INDEX FullTextOneProp1 FOR (n:FullTextOne) ON EACH [n.prop1]");
+        db.executeTransactionally("CALL db.awaitIndexes");
+
+        testCall(db, "CALL apoc.schema.properties.distinct($label, $key)",
+                map("label", FULL_TEXT_LABEL, "key", "prop1"), 
+                row -> assertEquals(Set.of("AA", "EE", "UU", "Ryan", "Michael"), Set.copyOf((List)row.get("value"))) 
+        );
         
+        testResult(db, SCHEMA_DISTINCT_COUNT_ORDERED,
+                map("label", FULL_TEXT_LABEL, "key", "prop1"),
+                res -> {
+                    extractedFullTextFullTextOneProp1(res);
+                    assertFalse(res.hasNext());
+                });
+
+        db.executeTransactionally("DROP INDEX FullTextOneProp1");
+    }
+    
+    @Test(timeout = 5000L)
+    public void testWithDifferentIndexesAndSameLabelProp() throws Exception {
+        db.executeTransactionally("CREATE FULLTEXT INDEX FullTextOneProp1 FOR (n:FullTextOne) ON EACH [n.prop1]");
+        db.executeTransactionally("CREATE RANGE INDEX RangeProp1 FOR (n:FullTextOne) ON (n.prop1)");
         db.executeTransactionally("CALL db.awaitIndexes");
 
         testCall(db, "CALL apoc.schema.properties.distinct($label, $key)",
-                map("label", "Label1", "key", "prop1"),
-                row -> assertEquals(Set.of("AA", "EE", "UU", "Ryan", "Michael", "abcde", "qwerty"), Set.copyOf((List)row.get("value")))
+                map("label", FULL_TEXT_LABEL, "key", "prop1"),
+                row -> assertEquals(Set.of("AA", "EE", "UU", "Ryan", "Michael"), Set.copyOf((List)row.get("value")))
         );
 
-        testResult(db, "CALL apoc.schema.properties.distinctCount($label, $key) " +
-                        "YIELD label, key, value, count RETURN * ORDER BY label",
-                map("label", "Label1", "key", "prop1"),
+        // in this case the procedure returns equal pairs of rows because we have 2 different indexes
```

Comment:
This might be surprising for a user. Do you think we should try to filter out equal lines or should we just document it?

Reply:
Yes, maybe it makes more sense to remove the duplicates to avoid any doubts. Updated